src/app/task/task.service.ts

```
@@ -4,94 +4,76 @@ import {
     status: TaskStatus,
     error?: string,
-  ): Promise<TaskEntity | null> {
-    if (!taskId || !status) {
-      throw new BadRequestException('Task ID and status are required');
+  ): Promise<TaskEntity> {
+    if (!status) {
+      throw new BadRequestException('Status is required');
     }
 
-    await this.taskRepo.updateTaskStatus(taskId, status, error);
+    const task = await this.findTaskOrThrow(taskId);
```

ESLint

[error] 'task' is assigned a value but never used.

(@typescript-eslint/no-unused-vars)